redfish-core/lib/log_services.hpp

```diff
@@ -1281,6 +1282,16 @@ inline void createDump(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
         }
         if ((*oemDiagnosticDataType).starts_with("Resource"))
         {
+            // Check if service user is requesting the dumps
+            // and then proceed with dump collection
+            std::string roleId = getRoleIdFromPrivilege(req.session->userRole);
+            if (roleId != "OemIBMServiceAgent" && roleId != "Administrator")
+            {
+                BMCWEB_LOG_WARNING(
+                    "Request to create resource dump failed. User is not a service/admin user");
+                messages::insufficientPrivilege(asyncResp->res);
+                return;
+            }
```

Comment:
This check is redundant.  The Admin/service privilege was already checked via
requestRoutesBMCDumpCreate() BMCWEB_ROUTE(...).privileges(redfish::privileges::postLogService).
You will need the RoleId for the check below.
The code is okay, but technically this should check if the session has privileges({{"OemIBMPerformService"}}),
example code in https://github.com/ibm-openbmc/bmcweb/blob/1110/redfish-core/lib/account_service.hpp#L1712

Reply:
> This check is redundant.

Agree. Will remove it.

> but technically this should check if the session has privileges({{"OemIBMPerformService"}}),

But this. Why is it any different from what I have used?
Basically, roleId is assigned based on the privilege (priv-admin or priv-oemserviceagent). Just thinking why this shouldn't be used.

Reply:
> > but technically this should check if the session has privileges({{"OemIBMPerformService"}}),
> 
> But this. Why is it any different from what I have used? Basically, roleId is assigned based on the privilege (priv-admin or priv-oemserviceagent). Just thinking why this shouldn't be used.

The Redfish authority model maps user accounts to Roles, maps Roles to Redfish Privileges, and maps Operations to Privileges.  Ideally all authority checks would be checking if the user's Privileges are enough to perform the privileges required by the Operation.  But the service user is very special, has the Restricted ServiceAgent  Role which has a special privilege OemIBMPerformService... so the check for Role has the same outcome as the check for privilege.

Reply:
So, the current change that I have still should be okay I believe.